### PR TITLE
Fixes issue with plural search terms

### PR DESCRIPTION
**Pull request by jimmynotjim**
Fixed issue where plural search terms were returning different results than singular terms

## Additions

- Adds ability to scrub plural terms in typeAndFilter plugin

## Removals

- None

## Changes

- None

## Testing

- fetch branch and navigate to `/contact-us` and attempt singular and plural search terms

## Review

- @schaferjh 
- @sebworks 
- @KimberlyMunoz 
- @anselmbradford 

## Preview

screen shot 2015-04-27 at 4 17 15 pm

## Notes

None

### CODE REVIEW

**Comment by sebworks** [comment]
@jimmynotjim, do you think users will be confused by the fact the we are changing the case and plurality in the results message? You search for 'Regulations' but the results message states 'results for "regulation".

**Comment by jimmynotjim** [comment]
@sebworks I thought about that but I wasn't sure if it'd be better to show what we're returning results for or what the user searched (scrubbing also removes extra spaces and words shorter than certain chars (3 in our case).

Maybe we can show a message like Google does "showing results for ___"? Not sure how that will fit with the upcoming success notification change.

Any opinions from the UX side? cc @juidai @schaferjh

**Comment by schaferjh** [comment]
@duelj and I were just discussing this. We are going to shift to a generic message, for this and other reasons. So it will read more along the lines of "## filtered results." I believe this will come through as part of the filter-action-highlighting that we discussed during FJ design review today.

TL;DR ok for now, a fix is in the works.

**Comment by anselmbradford** [comment]
Appears to be too much truncation going on with some words:
screen shot 2015-04-27 at 6 41 49 pm

Also, I think our writing style would put the period inside the quote.

**Comment by KimberlyMunoz** [comment]
I was worried about "mortgages" and "guides", "inquiries".

This is also a fun bug to consider.

image

File: static/js/jquery.type-and-filter.js [hunk 1]
```diff
@@ -190,6 +191,10 @@
         }).join(' ');
     };
 
+    $.fn.typeAndFilter.removePlurality = function( text ) {
+        return text.replace(/[e']s$/,'').replace(/([^aiou])s/, '$1');
```
Comment (anselmbradford) [inline static/js/jquery.type-and-filter.js]: Add space after comma `.replace(/[e']s$/, '')`

**Comment by jimmynotjim** [comment]
@KimberlyMunoz ah, good call with "fees". This is why I hate having to rely on a regex but I'll keep playing with it.

**Comment by jimmynotjim** [comment]
I think I have the matching issue fixed, can everyone fetch the changes and try it locally? Anything with "ies" will skip scrubbing entirely now. I thought about swapping it for "y" and searching for both but I think that's overly complicating things.

**Comment by KimberlyMunoz** [comment]
Looks good to me.

**Comment by jimmynotjim** [comment]
@anselmbradford result text is part of another issue that'll come later (see @schaferjh [comment above](https://github.com/cfpb/cfgov-refresh/pull/492#issuecomment-96814140))  so I'm considering it out of scope for this PR. If there's no other issues can I get a last :+1: from you?

**Comment by anselmbradford** [comment]
@jimmynotjim Is it a concern that this'll garble some words, e.g. "Buses" becoming "Buse"? That's just one word I can think of, but there may be others like that which are more applicable to our anticipated search keywords.

**Comment by jimmynotjim** [comment]
It sort of is, but "buse" will still match to "buses" in the text and like @schaferjh said, we're going to stop returning the string that the user searched for in the "results" title, so no one should really notice.

**Comment by anselmbradford** [comment]
I know the exact wording will be worked out elsewhere, but what do you think about stripping out the `for "{{ term }}"` bit in the result text in this PR? Otherwise, it's introducing code that will in some cases display a typo, which will be fixed at an indeterminate time in the future. Looks like just an edit to https://github.com/cfpb/cfgov-refresh/blob/flapjack/static/js/jquery.type-and-filter.js#L37-L38 and https://github.com/cfpb/cfgov-refresh/blob/flapjack/static/js/jquery.type-and-filter.js#L130 to remove the search term. I'm not sure if we have associated tests that check this or not, but since testing is still a bit scattered (https://github.com/cfpb/cfgov-refresh/issues/462) that may be okay to ignore for now.

Also, should this algorithm be developed further down the road, this seems like a good enhancement that could be added later: http://stackoverflow.com/questions/796412/how-to-turn-plural-words-singular

**Comment by jimmynotjim** [comment]
1. Removed
2. From the 2nd comment in that link:

    > The problem is that's based on the general rules, but English has (figuratively) a billion exceptions... What do you do with words like "fish", or "geese"?

    At some point we're crossing into diminishing returns and adding any more complexity than the regex I've suggested is probably where that line is. We're not Google, we're just trying to account for a few simple plural forms in use on this single page. 
 
As far as this PR, we good?

**Comment by anselmbradford** [comment]
> At some point we're crossing into diminishing returns and adding any more complexity than the regex 

Yeah, just the architecture of the top comment in how it chains the transformations is nice. So if we run into a word that got garbled going through and there were rules chained together, we could easily add it to an exceptions table. It's kind of set up that way already in the `scrubText` method.

> As far as this PR, we good?

:+1: :+1: :+1:

**Comment by anselmbradford** [comment]
Oh one last thing... I don't think https://github.com/cfpb/cfgov-refresh/blob/DF-1811-contact-filters/static/js/jquery.type-and-filter.js#L130 is needed any longer since the template doesn't have {{ term }} in it now.

**Comment by jimmynotjim** [comment]
@anselmbradford I left it in for now because it's still called for in the [base settings](https://github.com/cfpb/cfgov-refresh/blob/DF-1811-contact-filters/static/js/jquery.type-and-filter.js#L38) and I still feel this is outside the scope of this PR.

**Comment by anselmbradford** [comment]
Ahh I gotcha, didn't see that app.js overrides those defaults.
